Not compatible with ffmpeg 5

On ArchLinux, with `cargo install gifski --features=video`, rustc complains the following
```
error[E0599]: no method named `codec` found for struct `Stream` in the current scope
  --> /home/sb/.cargo/registry/src/github.com-1ecc6299db9ec823/gifski-1.6.6/src/bin/ffmpeg_source.rs:48:34
   |
48 |             let decoder = stream.codec().decoder().video().map_err(|e| format!("Unable to decode the codec used in the video: {}", e))?;
   |                                  ^^^^^ method not found in `Stream<'_>`

For more information about this error, try `rustc --explain E0599`.
error: failed to compile `gifski v1.6.6`, intermediate artifacts can be found at `/tmp/cargo-installuQxfHf`

Caused by:
  could not compile `gifski` due to previous error
```
But apparently that method [does exist](https://docs.rs/ffmpeg/0.3.0/ffmpeg/struct.Stream.html#method.codec).

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,7 +48,7 @@ scopeguard = "1.1.0"
 
 [dependencies.ffmpeg]
 package = "ffmpeg-next"
-version = "4.4.0"
+version = ">= 4.4.0, < 7"
 optional = true
 default-features = false
 features = ["codec", "format", "filter", "software-resampling", "software-scaling"]

--- a/src/bin/ffmpeg_source.rs
+++ b/src/bin/ffmpeg_source.rs
@@ -45,7 +45,9 @@ impl FfmpegDecoder {
             let filter_fps = self.rate.fps / self.rate.speed;
             let stream = self.input_context.streams().best(ffmpeg::media::Type::Video).ok_or("The file has no video tracks")?;
 
-            let decoder = stream.codec().decoder().video().map_err(|e| format!("Unable to decode the codec used in the video: {}", e))?;
+            let mut codec_context = ffmpeg::codec::context::Context::new();
+            codec_context.set_parameters(stream.parameters())?;
+            let decoder = codec_context.decoder().video().map_err(|e| format!("Unable to decode the codec used in the video: {}", e))?;
 
             let (dest_width, dest_height) = self.settings.dimensions_for_image(decoder.width() as _, decoder.height() as _);
 
@@ -103,8 +105,13 @@ impl FfmpegDecoder {
         .chain(std::iter::once(ffmpeg::Packet::empty()));
 
         for packet in packets {
-            let decoded = decoder.decode(&packet, &mut vid_frame)?;
-            if decoded {
+            decoder.send_packet(&packet)?;
+            loop {
+                match decoder.receive_frame(&mut vid_frame) {
+                    Ok(()) => (),
+                    Err(ffmpeg::Error::Other { errno: ffmpeg::error::EAGAIN }) | Err(ffmpeg::Error::Eof) => break,
+                    Err(e) => return Err(Box::new(e)),
+                }
                 filter.get("in").ok_or("ffmpeg format error")?.source().add(&vid_frame)?;
                 let mut out = filter.get("out").ok_or("ffmpeg format error")?;
                 let mut out = out.sink();

